### PR TITLE
Dataloader tidying

### PR DESCRIPTION
**Pull request by baykaner**
- removed text loader and basic text loader
- enforced AddData methods on all data loaders

### CODE REVIEW

File: libs/ml/include/ml/dataloaders/code2vec_context_loaders/context_loader.hpp [hunk 1]
```diff
@@ -88,7 +89,7 @@ class C2VLoader : public DataLoader<LabelType, DataType>
   umap_str_int function_name_counter();
   umap_str_int path_counter();
   umap_str_int word_counter();
-  std::vector<std::pair<SizeType, std::tuple<SizeType, SizeType, SizeType>>> data;
+  std::vector<std::pair<SizeType, std::tuple<SizeType, SizeType, SizeType>>> data_;
```
Comment (evsmithx) [inline libs/ml/include/ml/dataloaders/code2vec_context_loaders/context_loader.hpp]: Public member variable shouldn't end with underscore

File: libs/vm_modules/src/ml/dataloaders/dataloader.cpp [hunk 2]
```diff
@@ -53,9 +53,38 @@ VMDataLoader::VMDataLoader(VM *vm, TypeId type_id)
...
+    mode_   = DataLoaderMode::MNIST;
+    loader_ = std::make_shared<CommodityLoaderType>();
+  }
+  else
+  {
+    throw std::runtime_error("unknown dataloader mode");
+  }
+}
+
+// Ptr<VMDataLoader> VMDataLoader::Constructor(VM *vm, TypeId type_id)
```
Comment (MissingNO57) [inline libs/vm_modules/src/ml/dataloaders/dataloader.cpp]: Commented out code?

**Review by MissingNO57 — CHANGES_REQUESTED** [review]
Soo close

File: libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp [hunk 3]
```diff
@@ -245,6 +258,26 @@ class MNISTLoader : public DataLoader<LabelType, T>
...
+  void GetAtIndex(SizeType index, ReturnType &ret)
+  {
+    SizeType i{0};
+    auto     it = buffer_.second.at(0).begin();
+    while (it.is_valid())
+    {
+      *it = static_cast<DataType>(data_[index][i]) / DataType{256};
+      i++;
+      ++it;
+    }
```
Comment (evsmithx) [inline libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp:258]: This loop would be neater as "for (auto it : buffer_.second.at(0))" etc.

File: libs/vm_modules/src/ml/dataloaders/dataloader.cpp [hunk 4]
```diff
@@ -53,9 +53,38 @@ VMDataLoader::VMDataLoader(VM *vm, TypeId type_id)
...
+//{
+//  return new VMDataLoader(vm, type_id);
+//}
```
Comment (evsmithx) [inline libs/vm_modules/src/ml/dataloaders/dataloader.cpp]: leftover comments

**Review by evsmithx — DISMISSED** [review]

**Review by MissingNO57 — DISMISSED** [review]
LGTM

**Review by evsmithx — DISMISSED** [review]

File: libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp [hunk 5]
```diff
@@ -93,14 +96,31 @@ class MNISTLoader : public DataLoader<LabelType, T>
...
+
+  void Display(InputType const &data) const
   {
     for (SizeType i{0}; i < FIGURE_WIDTH; ++i)
     {
       for (SizeType j{0}; j < FIGURE_HEIGHT; ++j)
       {
 
-        std::cout << (data.At(j, i, 0) > typename T::Type(0.5) ? char(219) : ' ');
+        std::cout << (data.At(j, i, 0) > typename InputType::Type(0.5) ? char(219) : ' ');
```
Comment (evsmithx) [inline libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp]: I assume that char(219) is meant to print a black square? Apparently this is platform-dependent as the ASCII standard only goes up to 128. I'd suggest changing it to "#" (char(35)) or at least adding a comment!

**Review by evsmithx — DISMISSED** [review]
LGTM

**Review by MissingNO57 — DISMISSED** [review]
LGTM

**Review by MissingNO57 — APPROVED** [review]
LGTM

**Pull request merged**